fem/fespace.hpp

```diff
@@ -128,8 +135,36 @@ class FiniteElementSpace
 
    /// The element restriction operators, see GetElementRestriction().
    mutable OperatorHandle L2E_nat, L2E_lex;
+   /// The face restriction operators, see GetFaceRestriction().
+   using key_face = std::tuple<bool, ElementDofOrdering, FaceType, L2FaceValues>;
```

Comment:
The code may be a bit clearer (for people like me -- I had to look up what `std::get` does) if `key_face` is a simple struct